airflow-core/tests/unit/serialization/test_serde.py

```diff
@@ -308,6 +316,17 @@ def test_serder_dataclass(self):
         d = deserialize(e)
         assert i.x == getattr(d, "x", None)
 
+    def test_serder_pydantic_dataclass(self):
+        orig = PydanticDataclass(a=5, b="SerDe Pydantic Dataclass Test")
+        serialized = serialize(orig)
+        assert orig.__version__ == serialized[VERSION]
+        assert qualname(orig) == serialized[CLASSNAME]
+        assert serialized[DATA]
+
+        decoded = deserialize(serialized)
+        assert orig.a == getattr(decoded, "a", None)
+        assert orig.b == getattr(decoded, "b", None)
```

Comment:
```suggestion
        assert decoded.a == orig.a
        assert decoded.b == orig.b
```

dataclasses support equality